tuf/data/roles.go

```diff
@@ -274,7 +272,9 @@ func mergeStrSlices(orig, new []string) []string {
 	}
 	for _, e := range new {
 		if !have[e] {
-			orig = append(orig, e)
+			origCopy := make([]string, len(orig))
+			copy(origCopy, orig)
+			orig = append(origCopy, e)
```

Comment:
Move the first two lines above the loop, assign the append result into origCopy, then return origCopy (might also want to call it something else, like `merged`):

``` 
origCopy := make([]string, len(orig), len(orig) + len(new))
copy(origCopy, orig)
for _, e := range new {
    if !have[e] {
        origCopy = append(origCopy, e)
    }
}
return origCopy
```

This should be more efficient for large `new` slices